### ISSUE TITLE
feat: add option for case-insensitive search

### ISSUE THREAD

**Issue by aarondill** [mong8se/actually.vim#3]
## Problem
In my interactive shell, i set `nocaseglob` and `completion-ignore-case on`, which means I can tab-complete and use globs in a case insensitive manner.
```
> ls
File_I_wanted file_2
> nvim f<tab>
# results in `nvim file_`
```

When neovim opens, `actually` detects that I meant another file, but only offers `file_2`, when in reality, I wanted `File_I_wanted`.

### Solution
Add a setup option (`case_insensitive`?) to search for the file you meant insensitively.

**Comment by aarondill** [comment]
This *might* be as simple as setting `:h 'wildignorecase'` before calling `vim.fn.glob` and then restoring it after; but that would need further testing.

**Comment by aarondill** [comment]
Additionally, we could read the directory (dirname of file name), and use lua to find any matching results (`file:lower():find()`). This however would break a "feature"(?) in that failed globs don't search.
```shell
> shopt -u nullglob
> ls
file file2
> nvim *ls
# actually searches with `*ls*`. Is intended? Is this ever actually useful?
```

**Comment by mong8se** [comment]
Interesting, it's case insensitive by default for me because of this:

```
'fileignorecase' 'fic'	boolean	(default on for systems where case in file
                                 names is normally ignored)
			global
	When set case is ignored when using file names and directories.
	See 'wildignorecase' for only ignoring case when doing completion.

```

The intention is for it to be case insensitive.

**Comment by aarondill** [comment]
I run linux, so `'fileignorecase'` defaults to false

**Comment by mong8se** [comment]
I pushed a fix. Interestingly `fileignorecase` results in a sort of combined a-z which is what I'm used to as it was defaulting to true. But `wildignorecase` results in a list of files sorted A-Z followed by a-z. Not sure why.

Let me know if it works for you and also if the `[slug]` pathnames are fixed too.

**Comment by aarondill** [comment]
Both fixes seem to work great! Thank you!
I'm closing this issue, since it's been resolved by badf0b757776a2851452068a2a547de3c09c5550.
